tests/exceptions/exceptions_constructors.cpp

```diff
@@ -0,0 +1,302 @@
+      sycl::exception e(static_cast<int>(errcode), sycl::sycl_category(),
+                        what_arg_str);
+      check_exception(e, std_errc, sycl::sycl_category(), what_arg_str);
+    }
+    SECTION(
+        "exception(int ev, const std::error_category& ecat, const char* "
+        "what_arg)") {
+      sycl::exception e(static_cast<int>(errcode), sycl::sycl_category(),
+                        what_arg_str.c_str());
+      check_exception(e, std_errc, sycl::sycl_category(), what_arg_str);
+    }
+    SECTION("exception(int ev, const std::error_category& ecat)") {
+      sycl::exception e(static_cast<int>(errcode), sycl::sycl_category());
+      check_exception(e, std_errc, sycl::sycl_category());
+    }
+
+    SECTION(
+        "exception(context ctx, std::error_code ec, const std::string& "
+        "what_arg)") {
+      sycl::exception e(ctx, std_errc, what_arg_str);
+      check_exception(e, std_errc, sycl::sycl_category(), what_arg_str, ctx);
+    }
+    SECTION(
+        "exception(context ctx, std::error_code ec, const char* what_arg)") {
+      sycl::exception e(ctx, std_errc, what_arg_str.c_str());
+      check_exception(e, std_errc, sycl::sycl_category(), what_arg_str, ctx);
+    }
+    SECTION("exception(context ctx, std::error_code ec)") {
+      sycl::exception e(ctx, std_errc);
+      check_exception(e, std_errc, sycl::sycl_category(), ctx);
+    }
+
+    SECTION(
+        "exception(context ctx, int ev,const std::error_category& ecat, const "
+        "std::string& what_arg)") {
+      sycl::exception e(ctx, static_cast<int>(errcode), sycl::sycl_category(),
+                        what_arg_str);
+      check_exception(e, std_errc, sycl::sycl_category(), what_arg_str, ctx);
+    }
+    SECTION(
+        "exception(context ctx, int ev,const std::error_category& ecat, const "
+        "char* what_arg)") {
+      sycl::exception e(ctx, static_cast<int>(errcode), sycl::sycl_category(),
+                        what_arg_str.c_str());
+      check_exception(e, std_errc, sycl::sycl_category(), what_arg_str, ctx);
+    }
+    SECTION("exception(context ctx, int ev,const std::error_category& ecat)") {
+      sycl::exception e(ctx, static_cast<int>(errcode), sycl::sycl_category());
+      check_exception(e, std_errc, sycl::sycl_category(), ctx);
+    }
+
+    SECTION("exception(const exception& other)") {
+      sycl::exception e(std_errc);
+      sycl::exception copy(e);
+      check_exception(copy, errcode, sycl::sycl_category());
+    }
+    SECTION("operator=(const exception& other)") {
+      sycl::exception e(std_errc);
+      sycl::exception copy;
+      CHECK_NOTHROW((copy = e));
+      check_exception(copy, errcode, sycl::sycl_category());
+    }
+  }
+}
+
+#ifdef SYCL_BACKEND_OPENCL
+TEST_CASE("Constructors for sycl::exception with OpenCL error code",
+          "[exception]") {
+  const std::string what_arg_str = "test";
+  const sycl::context ctx;
+  if (ctx.get_backend() == sycl::backend::opencl) {
```

Comment:
I think it would be better if this was an early return for the opposite case:

```suggestion
  if (ctx.get_backend() != sycl::backend::opencl) {
    WARN("OpenCL backend is not supported on this device");
    return;
  }
  // Rest of code without extra indentation
```

Reply:
Applied